sycl/include/sycl/ext/oneapi/work_group_scratch_memory.hpp

```diff
@@ -0,0 +1,70 @@
+//==--- work_group_scratch_memory.hpp - SYCL group local memory extension --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
+#include <sycl/detail/sycl_builtins.hpp>      // for __sycl_allocateLocalMemory
+#include <sycl/exception.hpp>                 // for exception
+#include <sycl/ext/oneapi/properties/properties.hpp> // for properties
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi {
+namespace experimental {
+
+__SYCL_ALWAYS_INLINE
+inline void *get_work_group_scratch_memory() {
+#ifdef __SYCL_DEVICE_ONLY__
+  return __sycl_dynamicLocalMemoryPlaceholder();
+#else
+  throw sycl::exception(
+      sycl::errc::feature_not_supported,
+      "sycl_ext_oneapi_work_scratch_memory extension is not supported on host");
+#endif
+}
+
+// Property
+struct work_group_scratch_size
+    : ::sycl::ext::oneapi::experimental::detail::run_time_property_key<
+          work_group_scratch_size,
+          ::sycl::ext::oneapi::experimental::detail::WorkGroupStaticMem> {
+  // Runtime property part
+  constexpr work_group_scratch_size(size_t bytes) : size(bytes) {}
+
+  size_t size;
+};
+
+using work_group_scratch_size_key = work_group_scratch_size;
+
+template <> struct is_property_key<work_group_scratch_size> : std::true_type {};
+
+template <>
+struct is_property_value<work_group_scratch_size>
+    : is_property_key<work_group_scratch_size> {};
+
```

Comment:
```suggestion
```
This isn't necessary. Can you point where our instructions still mention these? I need to clean that up.

Reply:
Ah, ok. I think I've removed these steps from the how-to comments in `properties.hpp` but the extension itself leaked that and hasn't been updated yet. We'll follow up on that.